test/end-to-end/application.js

```diff
@@ -84,6 +88,39 @@ describe("Company application endpoint test", () => {
                 expect(created_application).toHaveProperty("submittedAt", mockCurrentDate);
             });
 
+            test("Should send an email to admin and to company user", async () => {
+                const application = {
+                    email: "test2@test.com",
+                    password: "password123",
+                    companyName: "Testing company",
+                    motivation: "This company has a very valid motivation because otherwise, the tests would not exist.",
+                };
+                const res = await request()
+                    .post("/apply/company")
+                    .send(application);
+
+                expect(res.status).toBe(HTTPStatus.OK);
+
+                const adminEmailOptions = NEW_COMPANY_APPLICATION_ADMINS(
+                    application.email, application.companyName, application.motivation);
+                const companyEmailOptions = NEW_COMPANY_APPLICATION_COMPANY(
+                    application.companyName, res.body._id);
+
+                expect(EmailService.sendMail).toHaveBeenNthCalledWith(1, {
+                    subject: adminEmailOptions.subject,
+                    text: adminEmailOptions.text,
+                    html: adminEmailOptions.html,
+                    to: config.mail_address
+                });
+
+                expect(EmailService.sendMail).toHaveBeenNthCalledWith(2, {
+                    subject: companyEmailOptions.subject,
+                    text: companyEmailOptions.text,
+                    html: companyEmailOptions.html,
+                    to: application.email
+                });
```

Comment:
Can these fail if they ever occur in a different order? I.e. is there a way to make this check regardless of the order the method is called in?

Reply:
They never failed so far, since they are sent "synchronously", one after the other. But I could use an .includes check on the calls array instead, since the order doens't really matter